### ISSUE TITLE
restic provides insufficient feedback if S3 password is missing

### ISSUE THREAD

**Issue by mdbooth** [restic/restic#4937]
Output of `restic version`
--------------------------
restic 0.16.5 compiled with go1.22.4 on linux/amd64

What backend/service did you use to store the repository?
---------------------------------------------------------
Ceph rados gateway (s3)

Problem description / Steps to reproduce
----------------------------------------

I have a wrapper for running backups:
```bash
#!/bin/sh

#exec restic --password-command "pass show restic/workstation" \
exec restic -p /tmp/restic \
        -r s3:rgw-backup.home.heisenbug.com/workstation/restic "$@"
```

Note that it normally uses the output of pass to supply the password, but as I'm debugging and I want it to be identical in both environments I'm using a password file.

This works fine from machine A:
```bash
> restic-rgw snapshots
repository 42b59b1c opened (version 1)
ID        Time                 Host           Tags        Paths
----------------------------------------------------------------------
...
```

Machine A and machine B are running identical versions of restic.

On machine B, where I am trying to restore this backup, it does not work. After an extremely long delay I get:

```
Fatal: unable to open config file: Stat: Access Denied.
Is there a repository at the following location?
s3:rgw-backup.home.heisenbug.com/workstation/restic
```

I understand from https://github.com/restic/restic/issues/1671 that this is just a bad error message which means: something credentials-related happened.

Running strace reveals that the extremely long delay is due to it attempting to contact the AWS metadata service. Machine A does not do this.

strace also reveals that, unlike Machine A, Machine B never attempts to open `/tmp/restic` to read the password.

Adding `-vv` adds no additional logging. strace is the logging of last resort, and apart from telling me that restic seems to ignore the argument `-p /tmp/restic` on machine B but not machine A, I have nothing to go on.

Expected behavior
-----------------

Honestly I'm expecting this issue to be environmental. Assuming that to be the case, I need restic to give me some logs. I need it to tell me why it is ignoring the `-p /tmp/restic`. I need it to log what it is doing so I don't need strace and google to determine that it's attempting to get credentials from AWS metadata.

Actual behavior
---------------

```
strace -f -eopenat -o /tmp/strace -- restic-rgw snapshots
```

I observe that the working machine reads /tmp/restic, but the non-working machine does not.

Do you have any idea what may have caused this?
-----------------------------------------------

I'm attempting to restore to a new machine, so any number of things could be different.  The new machine obviously doesn't have any stored state from previous restic invocations.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

**Comment by mdbooth** [comment]
The underlying reason was that I hadn't put s3 credentials on the target machine, as opposed to the restic password.

My immediate issue is resolved, but the problem of a lack of logging and comprehensible error messages remains.

**Comment by MichaelEischer** [comment]
The slowness while selecting the S3 credentials is already by solved in restic 0.17.0 via https://github.com/restic/restic/issues/4707 . In your case it would now fail with `no credentials found. Use '-o s3.unsafe-anonymous-auth=true' for anonymous authentication`.

**Comment by MichaelEischer** [comment]
Let's track improving the error messages in https://github.com/restic/restic/issues/1671 .
